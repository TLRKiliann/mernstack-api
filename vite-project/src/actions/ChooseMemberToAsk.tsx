import React from 'react'
import { UserType } from '../models/usertype'
import { ComputerType } from '../models/computerType'

interface ChooseMemberToAskProps {
  user: UserType
  computer: ComputerType
  handleAskUserPrivate: (e: React.MouseEvent<HTMLButtonElement>, id: number) => void
  addUserById: (e: React.MouseMove<HTMLButtonElement>, id: number) => void
}

const ChooseMemberToAsk: React.FC = (props: ChooseMemberToAskProps) => {
  return (
    <span 
      data-testid="chooseMemberId"
      className="span--saloon">
      {props.user?.mainroom === props.computer?.title ? (
        <div className="user--imgsaloon">
          <img
            width="100%"
            height="100%"
            src={props.user?.img} 
            className="img--saloon"
          />
          <p className="useronline">
          {props.user?.firstName} {props.user?.isConnected ? (
            <span
              className="useronline--iconconnectected"
            >
              ✔
            </span>
            ):(
            <span
              className="useronline--icondeconnectected"
            >
              ❌
            </span>
            ) 
          }
          </p>
          <span
            onClick={() => props.handleAskUserPrivate(props.user?.id)}
            className="askprivate--service"
            title="Invite to private chat"
          >
            ✉
          </span>

          <span key={props.user?.id} className="lastspan--online">
            <button
              onClick={() => props.addUserById(props.user?.id)}
              className="btn--lastspanonline"
              title="Add as Your Friend"
            >
              +
            </button>
          </span>

        </div>
        ) : null
      }
    </span>
  )
}

export default ChooseMemberToAsk