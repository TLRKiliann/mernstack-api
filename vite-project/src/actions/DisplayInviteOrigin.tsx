import React, { useState } from 'react'
import { UserType } from '../models/usertype'
import serviceConfirm from '../services/serviceConfirm'
import DisplayInvitationOtherUsr from '../components/DisplayInvitationOtherUsr'

//Other user receives invitation with info (sender & style of invitation)

interface DisplayProps {
  id: number
  roomStyle: string
  username: string
  initialSender: string
  roomName: string
  setDisplayConfirmInvite: React.Dispatch<React.SetStateAction<boolean>>
  setVersusUser: React.Dispatch<React.SetStateAction<Array<UserType>>>
  users: UserType[]
  refreshUsers: UserType[]
  setRefreshUsers: React.Dispatch<React.SetStateAction<Array<UserType>>>
}

const DisplayInviteOrigin: React.FC = (props: DisplayProps) => {
  const [isCheckInvite, setIsCheckInvite] = useState<boolean>(false)
  const [isNotCheckInvite, setIsNotCheckInvite] = useState<boolean>(false)

  const handleInvitedResponse = (e: React.MouseEvent<HTMLButtonElement>, roomName: string) => {
    if (isCheckInvite === true) {
      const newRoom: string = roomName
      const findSentMsg: UserType = props.refreshUsers?.find((u) => u?.sentMsg.length !== 0)
      const senderInvite: string = findSentMsg?.sentMsg
      const searchOtherUser: UserType = props.users?.find((u) => u?.firstName === senderInvite)
      props.setVersusUser(searchOtherUser)
      const user: UserType = props.users?.find((user) => user?.firstName === props.username)
      const id: number | null = user?.id;
      const confirmInvitFromOther: UserType = props.users?.find((user) => user.id === id)
      const changeReturnConf: object = {...confirmInvitFromOther, room: newRoom, signalRecieve: 0,
        returnConfirm: 1}

      serviceConfirm
        .updateResponseUser(id, changeReturnConf)
        .then(initialData => {
          props.setRefreshUsers(props.users?.map((user) => user?.id === id ? 
            {
              id: id,
              img: user.img,
              firstName: user.firstName,
              lastName: user.lastName,
              age: user.age,
              email: user.email,
              location: user.location,
              gender: user.gender,
              mainroom: user.mainroom,
              room: newRoom,
              isConnected: user.isConnected,
              signalRecieve: false,
              sentMsg: user.sentMsg,
              messagebox: user.messagebox,
              returnConfirm: true
            } : user
          ))
        })
        .catch((error) => {
          props.setRefreshUsers(props.users?.filter((u) => u.id !== id))
          console.log("Problem to confirm msg handleBothConfirmation...", error)
        })
      props.setDisplayConfirmInvite(true)
    } else {
      const roomStyleDefined: string = props.roomStyle
      const findSentMsg = props.refreshUsers?.find((u) => u?.sentMsg.length !== 0)
      const senderInvite: string = findSentMsg?.sentMsg
      const searchOtherUser = props.users?.find((u) => u?.firstName === senderInvite)
      const user = props.users?.find((user) => user?.firstName === props.username)
      const id: number | null = user?.id;
      const confirmInvitFromOther: UserType = props.users?.find((user) => user.id === id)
      const changeReturnConf: object = {...confirmInvitFromOther, room: roomStyleDefined, signalRecieve: 0,
        returnConfirm: 0}

      serviceConfirm
        .updateResponseUser(id, changeReturnConf)
        .then(initialData => {
          props.setRefreshUsers(props.users?.map((user) => user?.id === id ? 
            {
              id: id,
              img: user.img,
              firstName: user.firstName,
              lastName: user.lastName,
              age: user.age,
              email: user.email,
              location: user.location,
              gender: user.gender,
              mainroom: user.mainroom,
              room: roomStyleDefined,
              isConnected: user.isConnected,
              signalRecieve: false,
              sentMsg: user.sentMsg,
              messagebox: user.messagebox,
              returnConfirm: false
            } : user
          ))
        })
        .catch((error) => {
          props.setRefreshUsers(props.users?.filter((u) => u.id !== id))
          console.log("Problem to confirm msg handleBothConfirmation...", error)
        })
      setIsNotCheckInvite(false)
      props.setDisplayConfirmInvite(false)
    }
  }

  const handleRejectInvite = () => {
    setIsNotCheckInvite(!isNotCheckInvite)
  }

  const handleSwitchBox = () => {
    setIsCheckInvite(!isCheckInvite)
  }

  return (
    <DisplayInvitationOtherUsr
      id={props.id}
      username={props?.username}
      initialSender={props?.initialSender}
      roomName={props?.roomName}
      isCheckInvite={isCheckInvite}
      isNotCheckInvite={isNotCheckInvite}
      handleSwitchBox={handleSwitchBox}
      handleRejectInvite={handleRejectInvite}
      handleInvitedResponse={(e) => handleInvitedResponse(e, props.roomName)}
    />
  )
}

export default DisplayInviteOrigin