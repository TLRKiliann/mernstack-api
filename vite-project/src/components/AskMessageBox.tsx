import React, { useState } from 'react'
import { db_users } from '../models/db_users'
import { UserType } from '../models/usertype'
import './styleComponents/AskMessageBox.scss'


interface db_usersProps {
  db_users: Array<UserType>
}

interface AskMessageBoxProps {
  catchById: db_usersProps
  handleInvitation: (e: React.MouseEvent<HTMLButtonElement>) => void
  handleClose: (e: React.MouseEvent<HTMLButtonElement>) => void
}

const AskMessageBox: React.FC = (props: AskMessageBoxProps) => {
    
  return(
    <div key={props.catchById?.id} className="boolean--result">
      <div
        data-testid="asktestid" 
        className="form--invitation"
      >
        <label htmlFor="users" className="lbl--invite">
          Invite
          <img
            src={props.catchById?.img}
            width="50px"
            height="50px"
            className="img--invite"
            alt="no img panel ask invite"
          />
          <span className="span--invite">
            {props.catchById?.firstName}&nbsp;
          </span> to private chat :
        </label>

        <select 
          name="invite" 
          id="invite"
          value={props.form}
          onChange={props.handleInviteChoice}
        >
        {props.options.map((option) => (
          <option
            key={option.value}
            name="invite"
            value={option.value}
          >
            {option.label}
          </option>
        ))}
        </select> 
        
        <button
          onClick={props.handleInvitation}
          className="btn--invitation"
        >
          Invite
        </button>

        <button
          onClick={props.handleClose}
          className="btn--close"
        >
          X
        </button>
      </div>
    </div>
  )
}

export default AskMessageBox