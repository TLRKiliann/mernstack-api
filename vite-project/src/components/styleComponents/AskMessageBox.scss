/*Invitation message*/
$purple: rgba(195,87,116, 1);
$sandybrown: rgba(244,164,96, 1);
$whitesmoke: #f5f5f5;
$dark_moderate_red: #ae3c52;
$navy: #000080;
$font_junction: 'Junction', sans-serif;

.boolean--result {
  position: absolute;
  top: 0px;
  right: 0px;
  left: 0px;
  width: 550px;
  margin: auto;
  margin-top: 20%;
  display: flex;
  background: linear-gradient(180deg,
    $sandybrown, $purple);
  border-radius: 15px;
  z-index: 4;
  .form--invitation {
    padding-top: 20px;
    padding-bottom: 20px;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    .lbl--invite {
      margin-top: 20px;
      display: flex;
      flex-direction: row;
      align-items: center;
      font-family: $font_junction;
      font-size: 1.4rem;
      color: $purple;
      .img--invite {
        margin-left: 10px;
        border-radius: 50% 50%;
        object-fit: cover;
      }
      .span--invite {
        margin-left: 10px;
        color: #50bcfe;
        text-shadow: 2px 1px 1px $navy;
      }
    }
    .lbl--choicechat {
      margin-top: 20px;
      padding: 10px;
      display: flex;
      align-items: center;
      font-family: $font_junction;
      font-size: 1.4rem;
      .select--invite {
        padding-top: 3px;
        padding-bottom: 3px;
        font-size: 0.9rem;
        background: $sandybrown;
        border: 1px solid $purple;
        border-radius: 3px;
        color: $purple;
        .option--invite {
        }
      }
    }
    .btn--invitation {
      margin-top: 20px;
      padding: 5px 20px;
      font-size: 1.0rem;
      font-weight: bold;
      background: linear-gradient(180deg,
        $sandybrown, orangered, $dark_moderate_red);
      border: 1px solid $purple;
      border-radius: 7px;
      color: $whitesmoke;
      &:hover {
        background: $sandybrown;
        color: $dark_moderate_red;
      }
    }
    .btn--close {
      position: absolute;
      top: 10px;
      right: 10px;
      padding: 3px 7px;
      object-fit: cover;
      background: radial-gradient($sandybrown, $purple);
      border: 1px solid $sandybrown;
      border-radius: 100%;
      color: $purple;
      &:hover {
        background: $purple;
        color: $whitesmoke;
      }
    }
  }
}