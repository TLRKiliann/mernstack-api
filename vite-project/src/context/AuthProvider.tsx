import React, { createContext, ReactNode,
  useState, useContext } from 'react'

type Props = {
  children: ReactNode
}

type AuthContextProps = {
  auth: object
  setAuth: React.Dispatch<React.SetStateAction<object>>
  username: string
  setUserName: React.Dispatch<React.SetStateAction<string>>
  switchLogin: boolean
  setSwitchLogin: React.Dispatch<React.SetStateAction<boolean>>
  toggle: () => void
  eraseAll: () => void
}

const AuthContext = createContext({} as AuthContextProps);

export const useAuthLogin: React.FC = () => {
  return useContext(AuthContext)
}

export const AuthProvider: React.FC = ({ children }: Props) => {

  const [auth, setAuth] = useState<object>({});
  const [switchLogin, setSwitchLogin] = useState(true);

  const [username, setUserName] = useState("");

  const toggle = () => {
      setSwitchLogin(!switchLogin);
  };

  const eraseAll = () => {
      setUserName("");
      setAuth("");
  };

  return (
    <AuthContext.Provider value={{
      auth,
      setAuth,
      username,
      setUserName,
      switchLogin,
      setSwitchLogin,
      toggle,
      eraseAll
      }}
    >
      {children}
    </AuthContext.Provider>
  );
};