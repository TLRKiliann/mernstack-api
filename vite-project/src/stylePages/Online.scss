.saloon--byusers {
  position: absolute;
  top: 0px;
  right: 0px;
  left: 0px;
  min-height: 100vh;
  background: linear-gradient(180deg, 
    purple, orange);

  .title--services {
    margin-top: 80px;
    margin-left: 20px;
    font-family: "Junction", sans-serif;
    text-shadow: 2px 1px 1px #000;
    color: lightgrey;
  }

  .divsection--saloon {
    width: 80%;
    margin: auto;
    display: grid;
    grid-column-gap: 20px;
    grid-row-gap: 20px;
    grid-template-columns: 1Fr 1Fr 1Fr 1Fr;
    grid-template-rows: 1Fr 1Fr 1Fr;
    margin-bottom: 40px;

    .titlebyroom {
      margin: auto;
      padding: 7px 0px;
      font-family: "Junction";
      text-align: center;
      background: purple;
      border-bottom: 1px solid slategray;
      border-radius: 15px 15px 0px 0px;
      color: lightgrey;
    }

    .section--saloon {
      background: #383838;
      border: none;
      border-radius: 15px;
      box-shadow: 0 0 4px #000;

      .span--saloon {
        display: flex;
        background: #242424;
        font-family: "DejaVu Sans";
        font-size: 18px;
        color: #50bcfe;
      }

      .user--imgsaloon {
        width: 100%;
        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: flex-start;
        border-bottom: 1px solid slategray;
      }

      .img--saloon {
        margin: 0px 10px;
        border-radius: 50% 50%;
        object-fit: cover;
      }

      .askprivate--service {
        position: absolute;
        width: 15%;
        display: flex;
        justify-content: right;

        font-size: 22px;
        cursor: pointer;
        color: orange;
        z-index: 3;

        &:hover {
          color: cyan;
        }
      }

      .lastspan--service {
        position: absolute;
        width: 18%;  
        display: flex;
        align-items: center;

        justify-content: right;
        font-size: 14px;
        font-family: "DejaVu Sans";
        font-weight: thin;
        color: orange;

        .btn--lastspanservice {
          margin-left: 10px;
          padding: 3px 10px;
          cursor: pointer;
          background: #606060;
          border-radius: 50% 50%;
          font-size: 1.0rem;
          font-weight: bold;
          color: #50bcfe;

          &:hover {
            background: #707070;
            color: lightgreen;
          }
        }
      }
    }
  }
}