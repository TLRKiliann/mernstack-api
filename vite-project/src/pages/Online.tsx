import React, { useState, useEffect } from 'react'
import { useNavigate } from 'react-router-dom'
import { useAuthLogin } from '../context/AuthProvider'
import { db_users } from '../models/db_users'
import { UserType } from '../models/usertype'
import { db_computers } from '../models/db_computers'
import { computerType } from '../models/computerType'
import AskMessageBox from '../components/AskMessageBox'
import ResponsePrivateMsg from '../components/ResponsePrivateMsg'
import '../stylePages/Online.scss'

type DataAllType = {
  db_users: UserType[]
  db_computers: computerType[]
}

type Field = {
  value?: sting
  label?: string
}

type Form = {
  invite: Field
}

const options: Field[] = [
  {
    label: "Private",
    value: "Private",
  },
  {
    label: "Rdv",
    value: "Rdv",
  },
  {
    label: "Briefing",
    value: "Briefing",
  }
]

const Online: React.FC = () => {

  const Navigate = useNavigate()

  const [form, setForm] = useState<Form>({
    invite: {value: 'Private'}
  })

  const [users, setUsers] = useState<Array<DataAllType>>([])
  const [computers, setComputers] = useState<Array<computerType>>([])
  const [group, setGroup] = useState<Array<string | number | boolean>>([])
  const [catchById, setCatchById] = useState<Array<UserType>>([])
  
  //console.log(catchById, "catchById")

  const [switchAsk, setSwitchAsk] = useState<boolean>(false)
  const [switchResponse, setSwitchResponse] = useState<boolean>(false)

  const { setOtherUser, setTweekGroup } = useAuthLogin();

  useEffect(() => {
    setUsers(db_users)
    setComputers(db_computers)
  }, [])

  const addUserById = (id: number) => {
    const userTodAdd = users.find(user => user.id === id)
    const verifyAlreadyExist = group.find(g => g.id === id)
    if (verifyAlreadyExist === userTodAdd) {
      console.log("This user already exists")
      setGroup(group.filter(g => g.id !== id))
    } else {
      console.log("Ok, new user accepted")
      setGroup([...group, userTodAdd])
      setTweekGroup([...group, userTodAdd])
      alert(`User ${userTodAdd.firstName} added to your group
        (look at your Profile !)`)
    }
  }

  const handleInviteChoice = (e: React.ChangeEvent<HTMLInputElement>): void => {
    const fieldName: string = e.target.name;
    const fieldValue: string | number  = e.target.value;
    const newField: Field = {[fieldName]: {value: fieldValue}}

    setForm({...form, ...newField});
  }

  const handleAskUserPrivate = (id: number) => {
    const catchUser = users?.find(user => user.id === id)
    setCatchById(catchUser)
    setSwitchAsk(!switchAsk)
  }

  const handleInvitation = (e: React.FormEvent<HTMLFormElement>): void => {
    e.preventDefault()
    //console.log(catchById, "catchUser")
    setOtherUser(catchById)
    setSwitchAsk(!switchAsk)
    setSwitchResponse(!switchResponse)
    //Navigate('/computerroom/privatemessage')
    
    /*const sendMessage = users?.find(user => user === catchById)
    console.log(sendMessage, "sendMessage")*/

    /*serviceRouting
      .getAllMembers(sendMessage, id)
      .then(initialData => {
        setUsers(initialData)
      })
      .catch((error) => {
        console.log("error", error)
      })
    */
  }

  const handleClose = () => {
    setSwitchAsk(false)
  }

  const handleCloseResponse = () => {
    setSwitchResponse(false)
  }

  const handleCloseConfirm = () => {
    setConfirmRequest(false)
  }

  return(
    <div className="saloon--byusers">

      <div className="title--services">
        <h1>Rooms</h1>
      </div>

      <div className="divsection--saloon">

      {switchAsk &&
        <AskMessageBox
          catchById={catchById}
          form={form.invite.value}
          options={options}
          handleInviteChoice={(e) => handleInviteChoice(e)}
          handleInvitation={(e) => handleInvitation(e)}
          handleClose={handleClose}
        />
      }

      {switchResponse && 
        <ResponsePrivateMsg
          form={form.invite.value}
          catchById={catchById}
          handleCloseResponse={handleCloseResponse}
        />
      }

      {Object.values(computers)?.slice(0, 12).map(computer =>
        <section key={computer.id} className="section--saloon">
          <h3 className="titlebyroom">{computer.title}</h3>

          {Object.values(users)?.map(user =>
            <span key={user.id} className="span--saloon">
              {user.mainroom === computer.title ? (
                <div className="user--imgsaloon">
                  <img
                    width="40px"
                    height="40px"
                    src={user.img} 
                    className="img--saloon"
                  />
                  <p style={{fontSize: '16px'}}>
                  {user.firstName} {user.isConnected ? (
                    <span
                      style={{
                        color: 'lightgreen',
                        marginLeft: "5px"
                      }}
                    >
                      ✔
                    </span>
                    ):(
                    <span
                      style={{
                        color: 'red',
                        marginLeft: "5px",
                        fontSize: "18px"
                      }}
                    >
                      ❌
                    </span>
                    ) 
                  }
                  </p>
                  <span
                    onClick={() => handleAskUserPrivate(user.id)}
                    className="askprivate--service"
                    title="Invite to private chat"
                  >
                    ✉
                  </span>

                  <span key={user.id} className="lastspan--online">
                    <button
                      onClick={() => addUserById(user.id)}
                      className="btn--lastspanonline"
                      title="Add as Your Friend"
                    >
                      +
                    </button>
                  </span>

                </div>
                ) : null
              }
            </span>
            )}
        </section>
      )}
      </div>
    </div>
  )
}

export default Online