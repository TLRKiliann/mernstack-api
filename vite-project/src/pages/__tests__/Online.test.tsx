import React from "react"
import { useNavigate, useParams, MemoryRouter } from 'react-router-dom'
import { useAuthLogin } from '../../context/AuthProvider'
import { screen, render } from '@testing-library/react'
import { test, expect, vi, beforeEach } from 'vitest'
import '@testing-library/jest-dom'
import { create } from 'react-test-renderer'
import Online from "../Online.tsx"
import addUserById from '../Online'
import handleInviteChoice from '../Online'
import handleAskUserPrivate from '../Online'
import handleInvitation from '../Online'
import handleClose from '../Online'
import handleCloseResponse from '../Online'
import AskMessageBox from '../Online'
import DisplayInviteOrigin from '../Online'
import ConfirmationOrigin from '../Online'
import ChooseMemberToAsk from '../Online'
//import {assert, assertType, expectTypeOf, beforeEach, afterEach, describe, expect, test, it, vi} from 'vitest'
//import "@testing-library/jest-dom/extend-expect"
//import "@testing-library/jest-dom"
//import { act } from 'react-dom/test-utils'

test("Online context testing", () => {
  beforeEach(async (context) => {
    // extend context
    context.foo = 'bar'
  })
  test('Online state', ({ foo }) => {
    console.log(foo) // 'bar'
  })
})

test("Online hook testing", () => {
  const mockedParams = vi.fn()
  vi.mock('react-router-dom', () => ({
    ...vi.importActual('react-router-dom') as any,
    useParams: () => ({params: "room"})
  }))

  const mockedAuth = vi.fn()
  vi.mock('react-router-dom', () => ({
    ...vi.importActual('react-router-dom') as any,
    useAuthLogin: () => ({mockedAuth})
  }))
})

test('Online by test id', () => {
  render(<Online />)
  const mytestid = screen.getByTestId("divtestone")
  expect(mytestid).toBeInTheDocument()
})

test('Online MatchSnapShot test Online', () => {
  const treeOnline = create(<Online />)
  expect(treeOnline.toJSON()).toMatchSnapshot()
})

test('Online MatchSnapShot test ChooseMemberToAsk', () => {
  const treeChooseMemberToAsk = create(
    <ChooseMemberToAsk />
    )
  expect(treeChooseMemberToAsk.toJSON()).toMatchSnapshot()
})

//correct
test('Online MatchSnapShot test AskMessageBox', () => {
  const treeConfirm = create(
    <AskMessageBox />
  )
  expect(treeConfirm.toJSON()).toMatchSnapshot()
})

//correct
test('Online MatchSnapShot test ConfirmationOrigin', () => {
  const treeConfirm = create(
    <ConfirmationOrigin />
  )
  expect(treeConfirm.toJSON()).toMatchSnapshot()
})

test('Online addUserById to be defined', () => {
  const funcAddUser = addUserById
  expect(funcAddUser).toBeDefined()
})

test('Online addUserById return id', () => {
  const beverage = { id: 2 }
  const addUserById = vi.fn(beverage => {beverage})
  addUserById()
  expect(addUserById).toHaveReturned(2)
})

test('Online handleInviteChoice to be defined', () => {
  const funcInviteChoice = handleInviteChoice
  expect(funcInviteChoice).toBeDefined()
})

test('Online handleAskUserPrivate to be defined', () => {
  const funcAskUser = handleAskUserPrivate
  expect(funcAskUser).toBeDefined()
})

test('Online handleInvitation to be defined', () => {
  const funcInvite = handleInvitation
  expect(funcInvite).toBeDefined()
})

test('Online handleClose to be defined', () => {
  const funcClose = handleClose
  expect(funcClose).toBeDefined()
})

test('Online handleCloseResponse to be defined', () => {
  const funcCloseResp = handleCloseResponse
  expect(funcCloseResp).toBeDefined()
})

