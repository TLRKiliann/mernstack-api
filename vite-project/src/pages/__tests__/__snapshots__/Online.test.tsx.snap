// Vitest Snapshot v1

exports[`MatchSnapShot test AskMessageBox 1`] = `
<div
  className="saloon--byusers"
  data-testid="divtestone"
>
  <div
    className="title--services"
  >
    <h1>
      Rooms
    </h1>
  </div>
  <div
    className="divsection--saloon"
  />
  <div />
</div>
`;

exports[`MatchSnapShot test ChooseMemberToAsk 1`] = `
<div
  className="saloon--byusers"
  data-testid="divtestone"
>
  <div
    className="title--services"
  >
    <h1>
      Rooms
    </h1>
  </div>
  <div
    className="divsection--saloon"
  >
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Hard Disk - SSD
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        CPU
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        RAM
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Drivers
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Systemd
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Process
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Client-Server
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Protocols
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Request - Response
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Firewall
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Proxy - VPN
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Cyber-Security
      </h3>
    </section>
  </div>
  <div />
</div>
`;

exports[`MatchSnapShot test ConfirmationOrigin 1`] = `
<div
  className="saloon--byusers"
  data-testid="divtestone"
>
  <div
    className="title--services"
  >
    <h1>
      Rooms
    </h1>
  </div>
  <div
    className="divsection--saloon"
  >
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Hard Disk - SSD
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        CPU
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        RAM
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Drivers
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Systemd
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Process
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Client-Server
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Protocols
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Request - Response
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Firewall
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Proxy - VPN
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Cyber-Security
      </h3>
    </section>
  </div>
  <div />
</div>
`;

exports[`MatchSnapShot test Online 1`] = `
<div
  className="saloon--byusers"
  data-testid="divtestone"
>
  <div
    className="title--services"
  >
    <h1>
      Rooms
    </h1>
  </div>
  <div
    className="divsection--saloon"
  />
  <div />
</div>
`;

exports[`Online MatchSnapShot test AskMessageBox 1`] = `
<div
  className="saloon--byusers"
  data-testid="divtestone"
>
  <div
    className="title--services"
  >
    <h1>
      Rooms
    </h1>
  </div>
  <div
    className="divsection--saloon"
  />
  <div />
</div>
`;

exports[`Online MatchSnapShot test ChooseMemberToAsk 1`] = `
<div
  className="saloon--byusers"
  data-testid="divtestone"
>
  <div
    className="title--services"
  >
    <h1>
      Rooms
    </h1>
  </div>
  <div
    className="divsection--saloon"
  >
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Hard Disk - SSD
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        CPU
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        RAM
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Drivers
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Systemd
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Process
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Client-Server
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Protocols
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Request - Response
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Firewall
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Proxy - VPN
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Cyber-Security
      </h3>
    </section>
  </div>
  <div />
</div>
`;

exports[`Online MatchSnapShot test ConfirmationOrigin 1`] = `
<div
  className="saloon--byusers"
  data-testid="divtestone"
>
  <div
    className="title--services"
  >
    <h1>
      Rooms
    </h1>
  </div>
  <div
    className="divsection--saloon"
  >
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Hard Disk - SSD
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        CPU
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        RAM
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Drivers
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Systemd
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Process
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Client-Server
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Protocols
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Request - Response
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Firewall
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Proxy - VPN
      </h3>
    </section>
    <section
      className="section--saloon"
    >
      <h3
        className="titlebyroom"
      >
        Cyber-Security
      </h3>
    </section>
  </div>
  <div />
</div>
`;

exports[`Online MatchSnapShot test Online 1`] = `
<div
  className="saloon--byusers"
  data-testid="divtestone"
>
  <div
    className="title--services"
  >
    <h1>
      Rooms
    </h1>
  </div>
  <div
    className="divsection--saloon"
  />
  <div />
</div>
`;
